src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1844,16 +1847,45 @@ class TcgStatics {
     def moveList = []
     def labelList = []
     target.each {pcs ->
-      moveList.addAll pcs.topPokemonCard.moves
-      labelList.addAll pcs.topPokemonCard.moves.collect {"$pcs.name - $it.name" }
-    }
-    Move move = (choose(moveList, labelList, "Choose an attack to use as this attack.") as Move).shallowCopy()
-    move.energyCost = delegate.thisMove.energyCost
+      def newMoves = bg.em().activateGetter(new GetMoveList(pcs))
+      moveList.addAll newMoves
+      labelList.addAll newMoves.collect {"$pcs.name - $it.name" }
+    }
+    moveList.add "Skip"
+    labelList.add "End Turn (Skip)"
+    def choice = choose moveList, labelList, "Choose an attack to use as this attack."
+    if (choice instanceof String) return
+    Move move = (choice as Move).shallowCopy()
+    bc "$delegate.self copied $move.name"
+    if (!params.keepEnergyRequirement) move.energyCost = delegate.thisMove.energyCost
     // TODO: Why can't we just skip BetweenTurns if it is a sub attack?
-    def bef = blockingEffect(BETWEEN_TURNS)
+    def bef = null
+    if (!params.unblockEndTurn)
+      bef = blockingEffect(BETWEEN_TURNS)
     attack(move)
-    bc "$delegate.self copied $move.name"
-    bef.unregisterItself bg.em()
+    bef?.unregisterItself bg.em()
+  }
+
+  /**
+   * Copies an attack from another PokemonCardSet as an Ability Attack
+   * @param delegate onAttack delegate
+   * @param target A Closure with a call to return the current targets (ex: { all() } or { bench() }
+   */
+  static metronomeA(Object delegate, Closure target) {
+    delegate.getterA GET_MOVE_LIST, delegate.self, {holder->
+      def moves = [] as Set
+      moves.addAll holder.object
+      target.call().each {
+        if (it == holder.effect.target) return
+        if (it instanceof Card) {
+          moves.addAll it.moves
+        }
+        if (it instanceof PokemonCardSet) {
+          moves.addAll bg.em().activateGetter(new GetMoveList(it))
```

Comment:
I suspect this can cause endless loop if two pokemon target each other e.g. Arceus Omniscient. Perhaps only add `it.topPokemonCard?.moves`?

Reply:
sounds ugly.. then perhaps add a self.active guard at the top somewhere..

Reply:
Makes sense. I've added a check that the target is active, otherwise it returns without changing anything. Seems to work fine with 4 Arceus LV X on board.